pkg/config/config.go

```diff
@@ -181,8 +181,6 @@ func init() {
 	// Cloud Foundry
 	Datadog.SetDefault("cloud_foundry", false)
 	Datadog.SetDefault("bosh_id", "")
-	// APM
-	BindEnvAndSetDefault("apm_enabled", true) // this is to support the transition to the new config file
```

Comment:
Wondering if this might be required in the containerized environments. cc @xvello 

Should we maybe:
```
Datadog.BindEnv("apm_config.enabled", "DD_APM_ENABLED")
```

Logs configuration management seems like a decent source of inspiration.

Reply:
Indeed, `DD_APM_ENABLED` was the envvar for agent5 and we should try to keep it like this for agent6 to ease the transition.

Although, if the trace-agent does not use `pkg/config` and has its own logic to handle the `DD_APM_ENABLED` envvar in order to exit `0`, that's fine for us.

If the auxilary agents exit 0, they are disabled. This is how process-agent is handled.

Reply:
Yup I removed it because:

- `apm_enabled` was the wrong key
- it should be `apm_config.enabled` but as the trace-agent is itself reading the `DD_APM_ENABLED` env var, I guess we don't need it?